models/issue_xref.go

```diff
@@ -296,7 +296,7 @@ func CommentTypeIsRef(t CommentType) bool {
 // RefCommentHTMLURL returns the HTML URL for the comment that created this reference
 func (comment *Comment) RefCommentHTMLURL() string {
 	if comment.RefCommentID == 0 {
```

Comment:
Judging by the amount of time it took me to understand this code flow, what about adding a comment:
```suggestion
	// Edge case for when the reference is inside the title or the description of the referring issue
	if comment.RefCommentID == 0 {
```